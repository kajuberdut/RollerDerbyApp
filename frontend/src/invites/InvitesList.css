.Invites {
    margin-top: 150px;
}

.InvitesList {
    background-color: #d1d2d4;
    margin-top: 20px;
    width: 43%;
    margin-left: 12%;
}