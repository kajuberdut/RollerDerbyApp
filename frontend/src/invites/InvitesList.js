import React, { useState, useEffect } from "react";
import FastApi from "../Api";
import Loading from "../multiUse/loading/Loading"
import './InvitesList.css'
import { Card, CardBody, CardTitle } from "reactstrap";
// import TeamForm from "../forms/TeamForm";
import InviteComponent from "./InviteComponent";

/**
 * Display invites list page
 */

function InvitesList() {

    /** Set invites and is loading in state*/

    const [isLoading, setIsLoading] = useState(true);
    const [invites, setInvites] = useState([]);
    const user = JSON.parse(localStorage.getItem('user'));

  /** API get request for invites */

  async function getInvites() {

    try {
      let invites = await FastApi.getInvites(user.userId);
      console.log("users in inviteslist !!!!!!!!!!!!!!!  ", invites)
      setInvites(invites)
      setIsLoading(false)
    } catch (errors) {
      return { success: false, errors };
    }
  }

   /** Reloads invites when changes request for invites */

    useEffect(() => {
        getInvites();
    }, []);


  /** Display loading if API call is has not returned */

    if (isLoading) {
      return (
          <Loading />
      )
    }

  /** Render the cards for users*/

    const renderCards = () => {
      return (
        <div className="TeamList-RenderCards">
                {invites.map(invite => (
                  <InviteComponent invite={invite} key={"Invite-" + invite.inviteId} />           
                ))}
          </div>
        );
    }
  
  /** Render search bar and cards */

    return (
      <div className="Invites">
      {/* <SearchComponentUsers getUsers={getUsers} setUsers={setUsers}/> */}
      <Card className="InvitesList" style={{width: '400px'}}>
        <CardBody>
        <CardTitle><h1>Invites</h1></CardTitle>
        {invites && <div> {renderCards()} </div> }
        {invites.length === 0 && <div>No pending invites.</div>}
        </CardBody>
      </Card>
      </div>
    );

}

export default InvitesList;