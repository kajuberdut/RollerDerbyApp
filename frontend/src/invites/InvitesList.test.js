import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import InvitesList from "./InvitesList";
import { BrowserRouter } from "react-router-dom";

// * Passing as of 2/13/24
// ! have not figured out how to mock api request
// npm test InvitesList.test.js
// must be in frontend directory 


afterEach(() => {
  cleanup()
})


describe("Tests Invite List Page Renders", () => {

    test("invite component should render", () => {
      render(
        <BrowserRouter>
              <InvitesList />
        </BrowserRouter>,
      );
    }); 

    test('renders loading when get all invites has not resolved', () => {
    
        render(<InvitesList />);

        const loading = screen.getByTestId('loading-indicator');
        expect(loading).toBeInTheDocument();
      
     });
  
    it("matches snapshot", function() {
      const { asFragment } = render(
        <BrowserRouter>
              <InvitesList />
        </BrowserRouter>,
      );
        expect(asFragment()).toMatchSnapshot();
    });

})
