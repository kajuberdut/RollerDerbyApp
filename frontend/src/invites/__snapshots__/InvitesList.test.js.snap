// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Invite List Page Renders matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Loading"
    data-testid="loading-indicator"
  />
</DocumentFragment>
`;
